src/components/ModalBackupWords.js

```diff
@@ -96,58 +100,85 @@ class ModalBackupWords extends React.Component {
         this.props.updateWords(words);
         this.setState({ passwordSuccess: true, errorMessage: '' });
       } else {
-        this.setState({errorMessage: t`Invalid password`})
+        this.setState({ errorMessage: t`Invalid password` });
       }
     }
-  }
+  };
 
   /**
-   * Called when user clicks the button saying that he has saved the words already  
-   * So we select some words to do the validation and show validation component in the modal
+   * Called when user clicks the button saying that he has saved the words already
+   * Then, validation steps will ask the right word given an index between [1,24].
+   * As the words were shown previosly, the user should be able to find any word given an index.
    */
   handleWordsSaved = () => {
-    const wordsValidation = _.shuffle(this.props.words.split(' ')).slice(0, WORDS_VALIDATION);
-    this.setState({ showValidation: true, wordsValidation });
-  }
+    const wordsArray = this.props.words.split(' ');
+    const wordsToValidate = _.shuffle(wordsArray).slice(0, WORDS_VALIDATION);
+    const validationSteps = wordsToValidate.map((word, index) => {
+      const backupIndex = wordsArray.indexOf(word);
+      let optionsStartIndex = backupIndex - 2;
+      let optionsEndIndex = backupIndex + 2;
+
+      // If index is 0 or 1, startIndex would be negative
+      // So we set start to 0 and end to 4
+      if (optionsStartIndex < 0) {
+        optionsStartIndex = 0;
+        optionsEndIndex = optionsStartIndex + 4;
+      }
+
+      // If index is 22 or 23, endIndex would be greater than the max
+      // So we set to the max and decrease the startIndex
+      const maxIndex = wordsArray.length - 1;
+      if (optionsEndIndex > maxIndex) {
+        optionsEndIndex = maxIndex;
+        optionsStartIndex = maxIndex - 4;
+      }
+
+      const options = _.shuffle(
+        wordsArray.slice(optionsStartIndex, optionsEndIndex + 1)
+      );
+
+      return {
+        backupIndex: backupIndex + 1,
+        correctWord: word,
+        options,
+        done: false,
+        last: index === wordsToValidate.length - 1,
+      };
+    });
+    this.setState({ showValidation: true, validationSteps });
+  };
 
   /**
    * Called when user finishes the backup. We validate the backup and shows a success message, in case of success
    */
-  handleValidateBackup = () => {
-    if (this.state.chosenWords.length !== WORDS_VALIDATION) {
-      this.setState({ errorMessage: t`Invalid number of words` });
+  handleValidationStepOption = (option, validationStep) => {
+    validationStep.chosenOption = option;
+    if (option !== validationStep.correctWord) {
+      this.setState({
+        validationStep,
+        errorMessage: (
+          <span>
+            {t`Wrong word. Please double check the words you saved.`}
+            <strong onClick={() => this.setState({ showValidation: false, errorMessage: '' })}>Click here to start the process again.</strong>
```

Comment:
This should have the translation tag as well

Reply:
Done